app/assets/javascripts/assignment_form.js

```diff
@@ -7,12 +7,13 @@ function importOptions(starterCodeFieldValue) {
   }
 }
 
-function removeErrorBox() {
-  errorBoxes = document.getElementsByClassName("error");
-  for (var i = 0; i < errorBoxes.length; i++) {
-    errorBox = errorBoxes[i]
-    if (errorBox.innerText == "Starter code repository is not a template repository. Make it a template repository to use template cloning.") {
-      errorBox.style.display = 'none';
-    }
-  }
+function removeErrorBox(el) {
+  parent = el.closest(".errored");
+
+  if (!parent) return;
+
+  errorBoxes = parent.getElementsByClassName("error");
```

Comment:
Similar to above we should add `const`, `let` or `var`. Also, what if we started leaning toward `querySelector` so that we can start moving (slowly) to more modern and consistent finders? If we do that we can select a single `errorBox`, we're only expecting one:

```suggestion
  var errorBox = parent.querySelector(".error");
```